js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -251,9 +259,13 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
     private async transact<TReturn>(callback: (wallet: Web3MobileWallet) => TReturn): Promise<TReturn> {
         const walletUriBase = this._authorizationResult?.wallet_uri_base;
         const config = walletUriBase ? { baseUri: walletUriBase } : undefined;
+        const currentConnectionGeneration = this._connectionGeneration;
         try {
             return await transact(callback, config);
         } catch (e) {
+            if (this._connectionGeneration !== currentConnectionGeneration) {
+                await new Promise(() => {}); // Never resolve.
```

Comment:
I've never seen this kind of pattern (infinitely hanging promise) so I'm not sure what this does in this context. If a promise never resolves, does stuff hang around without getting GC'd, or stuck in any kind of weird state?